multidim_threshold/search.py

```diff
@@ -0,0 +1,69 @@
+import numpy as np
+import multidim_threshold
+from collections import namedtuple
+import more_itertools
+
+project_along = namedtuple('project_along', 'root direc')
+
+
+new_hi = lambda pi, hi: pi.root + np.multiply(pi.direc, np.true_divide(np.array(hi) - np.array(pi.root), np.array(pi.direc)).min())
+midpoint = lambda a, b: np.true_divide(np.array(a) + np.array(b), 2)
+
+def project_along_axes(lo, mid):
+    return [project_along_i(lo[:], mid, i) for i in range(len(lo))]
+
+def project_along_i(lo, mid, i):
+    lo[i] = mid[i]
```

Comment:
not super into implicitly updating the lo state during this function and the above projection.
Any way we can avoid? (thinking about introducing a `lenses` dependency so that might do the trick.